test/OpenTelemetry.Tests/Implementation/Trace/Propagation/CompositePropagatorTest.cs

```diff
@@ -0,0 +1,193 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using OpenTelemetry.Context.Propagation;
+using Xunit;
+
+namespace OpenTelemetry.Tests.Implementation.Trace.Propagation
+{
+    public class CompositePropagatorTest
+    {
+        private const string TraceParent = "traceparent";
+        private static readonly string[] Empty = new string[0];
+        private static readonly Func<IDictionary<string, string>, string, IEnumerable<string>> Getter = (headers, name) =>
+        {
+            if (headers.TryGetValue(name, out var value))
+            {
+                return new[] { value };
+            }
+
+            return Empty;
+        };
+
+        private static readonly Action<IDictionary<string, string>, string, string> Setter = (carrier, name, value) =>
+        {
+            carrier[name] = value;
+        };
+
+        private readonly ActivityTraceId traceId = ActivityTraceId.CreateRandom();
+        private readonly ActivitySpanId spanId = ActivitySpanId.CreateRandom();
+
+        [Fact]
+        public void CompositePropagator_NullTextFormatList()
+        {
+            Assert.Throws<ArgumentNullException>(() => new CompositePropagator(null));
+        }
+
+        [Fact]
+        public void CompositePropagator_WithTraceContext()
+        {
+            var expectedHeaders = new Dictionary<string, string>
+            {
+                { TraceParent, $"00-{this.traceId}-{this.spanId}-01" },
+            };
+
+            var compositePropagator = new CompositePropagator(new List<ITextFormat>
+            {
+                new TraceContextFormat(),
+            });
+
+            var activityContext = new ActivityContext(this.traceId, this.spanId, ActivityTraceFlags.Recorded, traceState: null);
+            var carrier = new Dictionary<string, string>();
+            compositePropagator.Inject(activityContext, carrier, Setter);
+
+            Assert.Equal(expectedHeaders, carrier);
+
+            var ctx = compositePropagator.Extract(activityContext, expectedHeaders, Getter);
+            Assert.Equal(activityContext.TraceId, ctx.TraceId);
+            Assert.Equal(activityContext.SpanId, ctx.SpanId);
+
+            Assert.Empty(compositePropagator.Fields);
+        }
+
+        [Fact]
+        public void CompositePropagator_WithTraceContextAndB3Format()
+        {
+            var expectedHeaders = new Dictionary<string, string>
+            {
+                { TraceParent, $"00-{this.traceId}-{this.spanId}-01" },
+                { B3Format.XB3TraceId, this.traceId.ToString() },
+                { B3Format.XB3SpanId, this.spanId.ToString() },
+                { B3Format.XB3Sampled, "1" },
+            };
+
+            var compositePropagator = new CompositePropagator(new List<ITextFormat>
+            {
+                new TraceContextFormat(),
```

Comment:
should we repeat this test with the order reversed?

Reply:
sure. let me change.